src/providers/arcgisrest/qgsarcgisrestutils.cpp

```diff
@@ -964,6 +982,24 @@ QgsFeatureRenderer *QgsArcGisRestUtils::parseEsriRenderer( const QVariantMap &re
   return nullptr;
 }
 
+QString QgsArcGisRestUtils::parseEsriArcadeExpression( const QString &string )
```

Comment:
Better add some unit tests for this method -- it looks complex, and is likely only going to get more so!

Reply:
@nyalldawson , the testParseLabeling test covers all that's currently implemented in parseEsriArcadeExpression, is that satisfactory for now?